sunpy/coordinates/tests/test_wcs_utils.py

```diff
@@ -265,3 +268,103 @@ def test_self_observer():
     assert u.allclose(wcs.wcs.aux.hglt_obs, frame.lon.to_value(u.deg))
     assert u.allclose(wcs.wcs.aux.crln_obs, frame.lon.to_value(u.deg))
     assert u.allclose(wcs.wcs.aux.dsun_obs, frame.radius.to_value(u.m))
+
+
+@pytest.fixture
+def dkist_location():
+    return EarthLocation(*(-5466045.25695494, -2404388.73741278, 2242133.88769004) * u.m)
+
+
+@pytest.mark.remote_data
+def test_obsgeo_frame_mapping_cartesian(dkist_location, caplog):
+
+    obstime = Time("2021-05-21T03:00:00")
+    wcs = WCS(naxis=2)
+    wcs.wcs.ctype = ['HPLT', 'HPLN']
+    wcs.wcs.obsgeo = list(dkist_location.to_value(u.m).tolist()) + [0, 0, 0]
+    wcs.wcs.dateobs = obstime.isot
+
+    frame = solar_wcs_frame_mapping(wcs)
+
+    assert frame.observer is not None
+
+    assert frame.observer == SkyCoord(dkist_location.get_itrs(obstime)).transform_to('heliographic_stonyhurst').frame
+
+    assert not caplog.records
+
+
+@pytest.mark.remote_data
+def test_frame_mapping_obsgeo_spherical(dkist_location, caplog):
+
+    obstime = Time("2021-05-21T03:00:00")
+    location = dkist_location.get_itrs(obstime)
+    loc_sph = location.spherical
+    wcs = WCS(naxis=2)
+    wcs.wcs.ctype = ['HPLT', 'HPLN']
+    wcs.wcs.obsgeo = [0, 0, 0] + [loc_sph.lon.value, loc_sph.lat.value, loc_sph.distance.value]
+    wcs.wcs.dateobs = obstime.isot
+
+    frame = solar_wcs_frame_mapping(wcs)
+
+    assert frame.observer is not None
+
+    assert frame.observer == SkyCoord(location).transform_to('heliographic_stonyhurst').frame
+
+    assert not caplog.records
+
+
+# TODO: Remove all these tests after Astropy 5.0 when we can just import obsgeo_to_frame.
+def test_obsgeo_cartesian(dkist_location):
+    obstime = Time("2021-05-21T03:00:00")
+    wcs = WCS(naxis=2)
+    wcs.wcs.obsgeo = list(dkist_location.to_value(u.m).tolist()) + [0, 0, 0]
```

Comment:
```suggestion
    wcs.wcs.obsgeo[:3] = dkist_location.to_value(u.m).tolist()
```
Presumably `list(a.tolist())` wasn't achieving anything?

Reply:
> Presumably `list(a.tolist())` wasn't achieving anything?

Turns out it was:

```
>       wcs.wcs.obsgeo = dkist_location.to_value(u.m).tolist() + [0, 0, 0]
E       TypeError: can only concatenate tuple (not "list") to tuple
```

Thanks for that one numpy, why does `to_list()` return a `tuple`??!